var/spack/repos/builtin/packages/chai/package.py

```diff
@@ -32,29 +35,53 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
     variant('raja', default=False, description='Build plugin for RAJA')
     variant('benchmarks', default=False, description='Build benchmarks.')
     variant('examples', default=True, description='Build examples.')
+    variant('openmp', default=False, description='Build using OpenMP')
     # TODO: figure out gtest dependency and then set this default True
     # and remove the +tests conflict below.
     variant('tests', default=False, description='Build tests')
 
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
 
-    depends_on('blt@0.4.0:', type='build', when='@2.3.1:')
-    depends_on('blt@:0.3.6', type='build', when='@:2.3.0')
+    depends_on('blt@0.4.1:', type='build', when='@2.4.0:')
```

Comment:
Per our discussion, if you don't want BLT maintainers to have to change your packages if BLT@0.5.0 introduces a breaking change, maybe we should say that BLT@0.4.1 is the latest supported? That remark may not apply here, and I may just be "philosophizing" ;).

Reply:
Is it best to introduce that constraint when we see the breakage though. Otherwise, if BLT 0.5.0 comes out and it works, it's incorrect to have the constraint. The best solution would be a "preferred" version dependency that can be overridden.